src/deps/zcl_abaplint_deps_find.clas.abap

```diff
@@ -263,59 +398,141 @@ CLASS ZCL_ABAPLINT_DEPS_FIND IMPLEMENTATION.
     SORT lt_tadir BY ref_obj_type ASCENDING ref_obj_name ASCENDING.
     DELETE ADJACENT DUPLICATES FROM lt_tadir COMPARING ref_obj_type ref_obj_name.
 
-    DELETE lt_tadir WHERE ref_obj_type = 'FUGR'.
-    DELETE lt_tadir WHERE ref_obj_type = 'DTEL'.
-    DELETE lt_tadir WHERE ref_obj_type = 'SFSW'.
-    DELETE lt_tadir WHERE ref_obj_type = 'DEVC'.
-    DELETE lt_tadir WHERE ref_obj_type = 'SUSO'.
-    DELETE lt_tadir WHERE ref_obj_type = 'TYPE'.
-    DELETE lt_tadir WHERE ref_obj_type = 'TTYP'.
-    DELETE lt_tadir WHERE ref_obj_type = 'DOMA'.
-    DELETE lt_tadir WHERE ref_obj_type = 'XSLT'.
-    DELETE lt_tadir WHERE ref_obj_type = 'SHLP'.
-    DELETE lt_tadir WHERE ref_obj_type = 'SQLT'.
-
-    DATA ls_tadir LIKE LINE OF lt_tadir.
-    DATA lv_devclass TYPE devclass.
-    DATA lv_index TYPE i.
+    IF lines( lt_tadir ) = 0.
+      RETURN.
+    ENDIF.
+
+*
+* Remove entries already in collection
+*
+    IF lines( mv_results ) > 0.
+      LOOP AT lt_tadir INTO ls_tadir.
+        lv_index = sy-tabix.
+        READ TABLE mv_results WITH KEY ref_obj_type = ls_tadir-ref_obj_type
+                                       ref_obj_name = ls_tadir-ref_obj_name
+                              TRANSPORTING NO FIELDS.
+        IF sy-subrc = 0.
+          DELETE lt_tadir INDEX lv_index.
+        ENDIF.
+      ENDLOOP.
+    ENDIF.
 
+    IF lines( lt_tadir ) = 0.
+      RETURN.
+    ENDIF.
+*
+* Remove entries from own package (or sub packages)
+*
```

Comment:
this block made sense when the deps were serialized for another online repo. i believe now, we need to remove this block since all dependencies - including within the same package - must be passed to the abaplint server. no?

Reply:
This block is a performance improvement to ensure that we handle dependencies for an object only once.
The object is collecting centrally now (not by call as it was before) and will return the complete list at the end.